### ISSUE TITLE
Move libcxx tests to a mystikos dockerhub registry

### ISSUE THREAD

**Issue by salsal97** [deislabs/mystikos#162]
[libcxx tests ](https://github.com/deislabs/mystikos/blob/main/tests/libcxx/Makefile#L74) are currently being pulled from a container in the "sagoel" dockerhub registry, this container should be moved to be under a "mystikos" dockerhub registry since it is a part of our testing infrastructure. 
Building the libcxx tests takes a lot of time, so building them on every run might not be feasible.

**Comment by AevaOnline** [comment]
Thanks for bringing this up, @salsal97. I think we should create a dockerhub account for the project, so that test, infra, and demo components (like this) can be stored there, and also so that we can publish pre-built images to it to create a faster onramp for users.

**Comment by jxyang** [comment]
Agreed.

**Comment by praenubilus** [comment]
we should consider apply the same thing to time consuming tests/samples, e.g. dotnet.

**Comment by AevaOnline** [comment]
Should we create one docker image which contains all of the necessary test infrastructure / dependent components? 

This has some advantages over managing a growing set of different images for different test cases (less images to keep track of, only one publish job, only one image to debug when something breaks) ... but also could have some drawbacks (less test isolation, likely cross-test dependencies).

**Comment by salsal97** [comment]
There are advantages to that since we pull the image down only once, but  having smaller image sizes would be ideal since it does determine the size of your appdir. But with ext2 this isn't a major drawback. 

You risk all the suites that use the image, with a higher potential for error each time any of them have to be updated, as opposed to just working on a container specific to a suite. Test isolation does make it seem like it's not worth it. Each time we update the image as well it would take a much larger build time, especially the libcxx tests since they do take more than 60 minutes to build.

Once we have a mystikos registry it should be fairly straightforward to have different containers published under it.

**Comment by jxyang** [comment]
We have the mystikos registry now. https://hub.docker.com/orgs/mystikos/. It's a free account, so only 3 members are allowed. We might upgrade to a paid account when it's out of capacity.

Can you submit a PR to fix this issue, @salsal97 ?

**Comment by praenubilus** [comment]
> We have the mystikos registry now. https://hub.docker.com/orgs/mystikos/. It's a free account, so only 3 members are allowed. We might upgrade to a paid account when it's out of capacity.
> 
> Can you submit a PR to fix this issue, @salsal97 ?

Can we create an internal MCR registry, that will serve pipelines and test infrastructure better

**Comment by jxyang** [comment]
Does ACR have free organization account?

**Comment by AevaOnline** [comment]
We need these images to be publicly accessible (so an internal registry won't do) because they are required when anyone runs `make test`.

DockerHub would be better for visibility than ACR, and DeisLabs already has an organizational account there - https://hub.docker.com/u/deislabs

**Comment by salsal97** [comment]
Closing this issue since a mystikos dockerhub registry has been set up and the libcxx tests have been moved there
